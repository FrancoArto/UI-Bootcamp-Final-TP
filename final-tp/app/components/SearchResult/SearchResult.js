import React, { Component } from 'react';
import { View, Text } from 'native-base';
import styles from './searchResult.style';
import { FlatList, ActivityIndicator } from 'react-native';
import Tweet from '../Tweet/Tweet';

export default class SearchResult extends Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
    this.handleOnEndReached = this.handleOnEndReached.bind(this);
    this.goToUserProfile = this.goToUserProfile.bind(this)
    this.handleOnTweetPress = this.handleOnTweetPress.bind(this)
  }


  handleOnTweetPress(event) {
    this.props.onTweetPress(event)
  }

  handleOnEndReached() {
    this.props.handleOnEndReached();
  }

  goToUserProfile(event) {
    this.props.goToUserProfile(event);
  }

  renderItem({ item, index }) {
    return (
      <Tweet
        id_str={item.id_str}
        user={item.user}
        mainContent={item.text}
        uri={item.user.profile_image_url_https}
        accountName={item.user.screen_name}
        favorite_count={item.favorite_count}
        retweet_count={item.retweet_count}
        media={item.entities.media}
        navigationProp={this.props.navigationProp}
        created_at={item.created_at}
        goToUserProfile={this.goToUserProfile}
        onTweetPress={this.handleOnTweetPress}
      />
    )
  }


  render() {
    let searchText = this.props.searchText;
    searchText = searchText.replace(/%23/, '#');
    return (
      <View style={styles.container}>
        <Text>Search Results for: {searchText}</Text>
        <FlatList
          ref='searchResult'
          data={this.props.data}
          renderItem={this.renderItem}
          keyExtractor={(item, index) => index.toString()}
          onEndReachedThreshold={5}
          onEndReached={this.handleOnEndReached} />
      </View>
    );
  }
}
