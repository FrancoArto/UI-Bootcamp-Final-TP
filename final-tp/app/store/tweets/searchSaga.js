import { GET_SEARCHTWEETS_URL, GET_APPENDRESULTS_URL } from '../../api/apiUrls';
import { takeLatest, put, call, select, all } from 'redux-saga/effects';
import { FETCH_SEARCH_BEGIN, fetchSearchSuccess, fetchSearchError, FETCH_MORE_RESULTS_REQUEST, fetchMoreResultsSuccess, fetchMoreResultsFailure } from './searchTweetsActions';

const getSearchText = state => state.searchTweetsReducer.searchText
const getMaxId = state => state.searchTweetsReducer.data[state.searchTweetsReducer.data.length - 1].id
const count = 50


function* fetchSearchResults() {  
  try {
    const searchText = yield select(getSearchText)
    const response = yield call(fetch, GET_SEARCHTWEETS_URL(searchText))
    const data = yield call([response, "json"]);
    yield put(fetchSearchSuccess(data.statuses))
  } catch (er) {
    yield put(fetchSearchError(er))
  }
}

function* searchTweets() {
  yield takeLatest(FETCH_SEARCH_BEGIN, fetchSearchResults)   
}


function* fetchResultsToAppend() {
  try {
    const searchText = yield select(getSearchText)
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDRESULTS_URL(searchText, count, maxId))
    const results = yield call([response, "json"]);
    const resultArray = results.statuses
    const data = resultArray.slice(1)
    yield put(fetchMoreResultsSuccess(data))
  } catch (er) {
    yield put(fetchMoreResultsFailure(er))
  }
}

function* appendResults() {
  yield takeLatest(FETCH_MORE_RESULTS_REQUEST, fetchResultsToAppend)   
}



export function* searchSaga() {
  yield all([
    searchTweets(),
    appendResults()
  ])
}