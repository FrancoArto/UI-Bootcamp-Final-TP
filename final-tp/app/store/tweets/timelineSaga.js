import { GET_TIMELINE_URL, GET_APPENDTWEETS_URL } from '../../api/apiUrls';
import { FETCH_TIMELINE_BEGIN, fetchTimelineSuccess, fetchTimelineFailure, FETCH_MORE_TWEETS_REQUEST, fetchMoreTweetsSuccess, fetchMoreTweetsFailure } from './timeLineActions'
import { takeLatest, put, call, all, select } from 'redux-saga/effects';

const twitTimelineCount = 50;

function* fetchTimeline() {
  try {
    const response = yield call(fetch, GET_TIMELINE_URL(twitTimelineCount))
    const data = yield call([response, "json"]);
    yield put(fetchTimelineSuccess(data))
  } catch (er) {
    yield put(fetchTimelineFailure(er))
  }
}

function* loadTimeline() {
  yield takeLatest(FETCH_TIMELINE_BEGIN, fetchTimeline)   
}

const getMaxId = state => state.timeLineReducer.data[state.timeLineReducer.data.length - 1].id


function* fetchTweetsToAppend() {
  try {
    const maxId = yield select(getMaxId)
    const response = yield call(fetch, GET_APPENDTWEETS_URL(twitTimelineCount, maxId))
    const array = yield call([response, "json"]);
    const data = array.slice(1)
    yield put(fetchMoreTweetsSuccess(data))
  } catch (er) {
    yield put(fetchMoreTweetsFailure(er))
  }
}

function* appendTweets() {
  yield takeLatest(FETCH_MORE_TWEETS_REQUEST, fetchTweetsToAppend)   
}

export function* timelineSaga() {
  yield all([
    loadTimeline(),
    appendTweets()
  ])
}