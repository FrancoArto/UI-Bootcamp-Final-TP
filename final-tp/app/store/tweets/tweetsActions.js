export const FETCH_SEARCH_BEGIN = 'FETCH_SEARCH_BEGIN'
export const FETCH_SEARCH_SUCCESS = 'FETCH_SEARCH_SUCCESS'
export const FETCH_SEARCH_FAILURE = 'FETCH_SEARCH_FAILURE'
export const FETCH_MORE_RESULTS_REQUEST = 'FETCH_MORE_RESULTS_REQUEST'
export const FETCH_MORE_RESULTS_SUCCESS = 'FETCH_MORE_RESULTS_SUCCESS'
export const FETCH_MORE_RESULTS_FAILURE = 'FETCH_MORE_RESULTS_FAILURE'
export const FETCH_TIMELINE_BEGIN = 'FETCH_TIMELINE_BEGIN'
export const FETCH_TIMELINE_SUCCESS = 'FETCH_TIMELINE_SUCCESS'
export const FETCH_TIMELINE_FAILURE = 'FETCH_TIMELINE_FAILURE'
export const FETCH_MORE_TWEETS_REQUEST = 'FETCH_MORE_TWEETS_REQUEST'
export const FETCH_MORE_TWEETS_SUCCESS = 'FETCH_MORE_TWEETS_SUCCESS'
export const FETCH_MORE_TWEETS_FAILURE = 'FETCH_MORE_TWEETS_FAILURE'
export const FETCH_USERTIMELINE_BEGIN = 'FETCH_USERTIMELINE_BEGIN'
export const FETCH_USERTIMELINE_SUCCESS = 'FETCH_USERTIMELINE_SUCCESS'
export const FETCH_USERTIMELINE_FAILURE = 'FETCH_USERTIMELINE_FAILURE'
export const FETCH_SINGLETWEET_BEGIN = 'FETCH_SINGLETWEET_BEGIN'
export const FETCH_SINGLETWEET_SUCCESS = 'FETCH_SINGLETWEET_SUCCESS'
export const FETCH_SINGLETWEET_FAILURE = 'FETCH_SINGLETWEET_FAILURE'


export const fetchTweetsTimeline = () => ({
    type: FETCH_TIMELINE_BEGIN
})

export const fetchTimelineSuccess = (data) => ({
    type: FETCH_TIMELINE_SUCCESS,
    payload: data
})

export const fetchTimelineFailure = (error) => ({
    type: FETCH_TIMELINE_FAILURE,
    payload: error
})

export const fetchMoreTweets = () => ({
    type: FETCH_MORE_TWEETS_REQUEST,
})

export const fetchMoreTweetsSuccess = (data) => ({
    type: FETCH_MORE_TWEETS_SUCCESS,
    payload: data
})

export const fetchMoreTweetsFailure = (error) => ({
    type: FETCH_MORE_TWEETS_FAILURE,
    payload: error
})

export const fetchSearchBegin = (searchText) => ({
    type: FETCH_SEARCH_BEGIN,
    payload: searchText
});

export const fetchSearchSuccess = (twitsArray) => ({
    type: FETCH_SEARCH_SUCCESS,
    payload: twitsArray
});

export const fetchSearchError = error => ({
    type: FETCH_SEARCH_FAILURE,
    payload: error
});

export const fetchMoreResults = () => ({
    type: FETCH_MORE_RESULTS_REQUEST,
})

export const fetchMoreResultsSuccess = (data) => ({
    type: FETCH_MORE_RESULTS_SUCCESS,
    payload: data
})

export const fetchMoreResultsFailure = (error) => ({
    type: FETCH_MORE_RESULTS_FAILURE,
    payload: error
})

export const fetchUserTimelineBegin = () => ({
    type: FETCH_USERTIMELINE_BEGIN,
});

export const fetchUserTimelineSuccess = (twitsArray) => ({
    type: FETCH_USERTIMELINE_SUCCESS,
    payload: twitsArray
});

export const fetchUserTimelineError = error => ({
    type: FETCH_USERTIMELINE_FAILURE,
    payload: error
});

export const fetchSingleTweetBegin = (tweetId) => ({
    type: FETCH_SINGLETWEET_BEGIN,
    payload: tweetId
});

export const fetchSingleTweetSuccess = (tweet) => ({
    type: FETCH_SINGLETWEET_SUCCESS,
    payload: tweet
});

export const fetchSingleTweetError = error => ({
    type: FETCH_SINGLETWEET_FAILURE,
    payload: error
});