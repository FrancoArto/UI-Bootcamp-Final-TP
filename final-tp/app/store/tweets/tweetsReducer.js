import {
  FETCH_SEARCH_BEGIN,
  FETCH_SEARCH_SUCCESS,
  FETCH_SEARCH_FAILURE,
  FETCH_MORE_RESULTS_REQUEST,
  FETCH_MORE_RESULTS_SUCCESS,
  FETCH_MORE_RESULTS_FAILURE,
  FETCH_TIMELINE_BEGIN,
  FETCH_TIMELINE_SUCCESS,
  FETCH_TIMELINE_FAILURE,
  FETCH_MORE_TWEETS_REQUEST,
  FETCH_MORE_TWEETS_SUCCESS,
  FETCH_MORE_TWEETS_FAILURE,
  FETCH_USERTIMELINE_BEGIN,
  FETCH_USERTIMELINE_SUCCESS,
  FETCH_USERTIMELINE_FAILURE,
  FETCH_SINGLETWEET_BEGIN,
  FETCH_SINGLETWEET_SUCCESS,
  FETCH_SINGLETWEET_FAILURE
} from './tweetsActions'
import { tweetsInitialState } from '../initialState'


const tweetsReducer = (state = tweetsInitialState, action) => {
  switch (action.type) {
    case FETCH_SEARCH_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
        searchText: action.payload
      }
    }
    case FETCH_SEARCH_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        searchResults: action.payload
      }
    }
    case FETCH_SEARCH_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        searchText: null,
        searchResults: []
      }
    }
    case FETCH_MORE_RESULTS_REQUEST: {
      return {
        ...state,
        loading: false,
        error: null,
      }
    }
    case FETCH_MORE_RESULTS_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        searchResults: state.searchResults.concat(...action.payload),
      }
    }
    case FETCH_MORE_RESULTS_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        searchResults: []
      }
    }
    case FETCH_TIMELINE_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
      }
    }
    case FETCH_TIMELINE_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        data: action.payload,
      }
    }
    case FETCH_TIMELINE_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        data: [],
      }
    }
    case FETCH_MORE_TWEETS_REQUEST: {
      return {
        ...state,
        loading: false,
        error: null
      }
    }
    case FETCH_MORE_TWEETS_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        data: state.data.concat(...action.payload),
      }
    }
    case FETCH_MORE_TWEETS_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        data: [],
      }
    }
    case FETCH_USERTIMELINE_BEGIN: {
      return {
        ...state,
        loading: true,
        error: null,
      }
    }
    case FETCH_USERTIMELINE_SUCCESS: {
      return {
        ...state,
        loading: false,
        error: null,
        userTimeline: action.payload,
      }
    }
    case FETCH_USERTIMELINE_FAILURE: {
      return {
        ...state,
        loading: false,
        error: action.payload,
        userTimeline: [],
      }
    }
    case FETCH_SINGLETWEET_BEGIN: {
      return {
        ...state,
        loadingSingleTweet: true,
        error: null,
        singleTweetId: action.payload
      }
    }
    case FETCH_SINGLETWEET_SUCCESS: {
      return {
        ...state,
        loadingSingleTweet: false,
        error: null,
        singleTweet: action.payload,
      }
    }
    case FETCH_SINGLETWEET_FAILURE: {
      return {
        ...state,
        loadingSingleTweet: false,
        error: action.payload,
        singleTweet: {},
        singleTweetId: null
      }
    }
    default:
      return state;
  }
};
export default tweetsReducer;