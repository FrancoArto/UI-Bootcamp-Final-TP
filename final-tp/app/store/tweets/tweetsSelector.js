export const getSearchText = state => state.tweetsReducer.searchText
export const getMaxId = state => state.tweetsReducer.data[state.tweetsReducer.data.length - 1].id
export const getFilteredTweets = state => {
  const timeline = state.tweetsReducer.data
  return timeline.filter(tweet => {
    if ((state.settingsReducer.verified === true) && (tweet.user.verified === false)) {
      return false;
    } else if ((state.settingsReducer.following === true) && (tweet.user.following === false)) {
      return false;
    } else if ((state.settingsReducer.defaultInfo === true) && (tweet.user.default_profile === true)) {
      return false;
    } else if ((state.settingsReducer.withLink === true) && (tweet.entities.urls.length > 0)) {
      return false;
    } else if ((state.settingsReducer.withTruncatedText === true) && (tweet.truncated === true)) {
      return false;
    } else {
      return true;
    }
  })
}
export const getSingleTweetId = state => state.tweetsReducer.singleTweetId