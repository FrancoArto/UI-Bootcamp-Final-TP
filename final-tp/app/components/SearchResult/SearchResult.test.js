import React from 'react';
import {mount, shallow} from 'enzyme'
import SearchResult from './SearchResult';
import { tweetsArray } from '../../store/tweets/tweetsMock';


it('renders without crashing', () => {
  const searchText = 'Hola'
  const rendered = shallow(<SearchResult data={tweetsArray} searchText={searchText} />)
  expect(rendered).toMatchSnapshot();
});

it('handles tweet press', () => {
  let searchText = 'Hola'
  const onTweetPress = jest.fn()

  const wrapper = mount(<SearchResult 
    data={tweetsArray}
    searchText={searchText}
    onTweetPress={onTweetPress}/>)

  const tweet = wrapper.find('Tweet').first();
  tweet.props().onTweetPress();
  expect(onTweetPress).toHaveBeenCalled()
});

it('handles user profile link', () => {
  let searchText = 'Hola'
  const goToUserProfile = jest.fn()

  const wrapper = mount(<SearchResult 
    data={tweetsArray}
    searchText={searchText}
    goToUserProfile={goToUserProfile}/>)

  const tweet = wrapper.find('Tweet').first();
  tweet.props().goToUserProfile();
  expect(goToUserProfile).toHaveBeenCalled()
});

it('handles infinite scroll', () => {
  let searchText = 'Hola'
  const onEndReached = jest.fn()

  const wrapper = mount(<SearchResult 
    data={tweetsArray}
    searchText={searchText}
    handleOnEndReached={onEndReached}/>)

  const list = wrapper.find('FlatList').first();
  list.props().onEndReached();
  expect(onEndReached).toHaveBeenCalled()
});