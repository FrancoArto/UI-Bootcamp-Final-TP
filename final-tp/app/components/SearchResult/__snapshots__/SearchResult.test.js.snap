// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "flex": 0.85,
      "paddingLeft": 10,
    }
  }
>
  <Styled(Text)>
    Search Results for: 
    Hola
  </Styled(Text)>
  <FlatList
    data={
      Array [
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 17:26:35 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "twitter.com/i/web/status/1…",
                "expanded_url": "https://twitter.com/i/web/status/1075804669056823301",
                "indices": Array [
                  117,
                  140,
                ],
                "url": "https://t.co/Wlbbpw5YCQ",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 0,
          "favorited": false,
          "geo": null,
          "id": 1075804669056823300,
          "id_str": "1075804669056823301",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "Después del viaje de Juan Darthés a Brasil, Sabrina Cartabia (Abogada de Thelma Fardin): \\"Todas las denuncias de vi… https://t.co/Wlbbpw5YCQ",
          "truncated": true,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Mar 23 12:35:07 +0000 2009",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Noticias las 24 horas, por el canal líder en audiencia y credibilidad de Argentina",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "tn.com.ar",
                    "expanded_url": "http://www.tn.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/TX6TF1nOUs",
                  },
                ],
              },
            },
            "favourites_count": 1,
            "follow_request_sent": false,
            "followers_count": 4677890,
            "following": true,
            "friends_count": 124,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 25992212,
            "id_str": "25992212",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 9083,
            "location": "Argentina",
            "name": "TN - Todo Noticias",
            "notifications": false,
            "profile_background_color": "666666",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/25992212/1533675118",
            "profile_image_url": "http://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_link_color": "D51D2E",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "314BB9",
            "profile_text_color": "000000",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "todonoticias",
            "statuses_count": 84323,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/TX6TF1nOUs",
            "utc_offset": null,
            "verified": true,
          },
        },
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 17:26:21 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "bit.ly/2EFC2Hg",
                "expanded_url": "http://bit.ly/2EFC2Hg",
                "indices": Array [
                  86,
                  109,
                ],
                "url": "https://t.co/QYvzBj9SIU",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 1,
          "favorited": false,
          "geo": null,
          "id": 1075804612601569300,
          "id_str": "1075804612601569281",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"https://about.twitter.com/products/tweetdeck\\" rel=\\"nofollow\\">TweetDeck</a>",
          "text": "La búsqueda de Sofía Herrera: esta es una de las casas allanadas durante el operativo
https://t.co/QYvzBj9SIU",
          "truncated": false,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Mar 23 12:35:07 +0000 2009",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Noticias las 24 horas, por el canal líder en audiencia y credibilidad de Argentina",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "tn.com.ar",
                    "expanded_url": "http://www.tn.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/TX6TF1nOUs",
                  },
                ],
              },
            },
            "favourites_count": 1,
            "follow_request_sent": false,
            "followers_count": 4677890,
            "following": true,
            "friends_count": 124,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 25992212,
            "id_str": "25992212",
            "is_translation_enabled": true,
            "is_translator": false,
            "lang": "es",
            "listed_count": 9083,
            "location": "Argentina",
            "name": "TN - Todo Noticias",
            "notifications": false,
            "profile_background_color": "666666",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/25992212/1533675118",
            "profile_image_url": "http://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/962300034661396481/OzHK6TZr_normal.jpg",
            "profile_link_color": "D51D2E",
            "profile_sidebar_border_color": "FFFFFF",
            "profile_sidebar_fill_color": "314BB9",
            "profile_text_color": "000000",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "todonoticias",
            "statuses_count": 84323,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/TX6TF1nOUs",
            "utc_offset": null,
            "verified": true,
          },
        },
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 17:22:43 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "twitter.com/i/web/status/1…",
                "expanded_url": "https://twitter.com/i/web/status/1075803698432016384",
                "indices": Array [
                  117,
                  140,
                ],
                "url": "https://t.co/6oJqb91xdq",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 175,
          "favorited": false,
          "geo": null,
          "id": 1075803698432016400,
          "id_str": "1075803698432016384",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 32,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "D'Onofrio reveló un gesto solidario del Pity Martínez: \\"Un día vino a la Fundación y dijo: 'Tomen, esta plata es pa… https://t.co/6oJqb91xdq",
          "truncated": true,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Mon Feb 08 20:26:20 +0000 2010",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Diario deportivo | Desde 1996, el primero y único del país | Compartimos la pasión",
            "entities": Object {
              "description": Object {
                "urls": Array [],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "Ole.com.ar",
                    "expanded_url": "http://www.Ole.com.ar",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/33Fqv7RRKh",
                  },
                ],
              },
            },
            "favourites_count": 617,
            "follow_request_sent": false,
            "followers_count": 2576528,
            "following": true,
            "friends_count": 156,
            "geo_enabled": false,
            "has_extended_profile": false,
            "id": 112523709,
            "id_str": "112523709",
            "is_translation_enabled": false,
            "is_translator": false,
            "lang": "es",
            "listed_count": 6498,
            "location": "Argentina",
            "name": "Diario Olé",
            "notifications": false,
            "profile_background_color": "000000",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme1/bg.png",
            "profile_background_tile": false,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/112523709/1544453053",
            "profile_image_url": "http://pbs.twimg.com/profile_images/1014198717891653633/J8RZJyum_normal.jpg",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/1014198717891653633/J8RZJyum_normal.jpg",
            "profile_link_color": "669900",
            "profile_sidebar_border_color": "9AB154",
            "profile_sidebar_fill_color": "FFFFFF",
            "profile_text_color": "5E5E5E",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "DiarioOle",
            "statuses_count": 151471,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/33Fqv7RRKh",
            "utc_offset": null,
            "verified": false,
          },
        },
        Object {
          "contributors": null,
          "coordinates": null,
          "created_at": "Thu Dec 20 17:22:02 +0000 2018",
          "entities": Object {
            "hashtags": Array [],
            "symbols": Array [],
            "urls": Array [
              Object {
                "display_url": "lacapitalmdp.com/confirman-proc…",
                "expanded_url": "https://www.lacapitalmdp.com/confirman-procesamiento-con-prision-preventiva-para-cristina-kirchner-y-de-vido/",
                "indices": Array [
                  82,
                  105,
                ],
                "url": "https://t.co/cSqmPjYJ9g",
              },
            ],
            "user_mentions": Array [],
          },
          "favorite_count": 1,
          "favorited": false,
          "geo": null,
          "id": 1075803526738206700,
          "id_str": "1075803526738206721",
          "in_reply_to_screen_name": null,
          "in_reply_to_status_id": null,
          "in_reply_to_status_id_str": null,
          "in_reply_to_user_id": null,
          "in_reply_to_user_id_str": null,
          "is_quote_status": false,
          "lang": "es",
          "place": null,
          "possibly_sensitive": false,
          "possibly_sensitive_appealable": false,
          "retweet_count": 0,
          "retweeted": false,
          "source": "<a href=\\"http://twitter.com\\" rel=\\"nofollow\\">Twitter Web Client</a>",
          "text": "Confirman procesamiento con prisión preventiva para Cristina Kirchner y De Vido - https://t.co/cSqmPjYJ9g",
          "truncated": false,
          "user": Object {
            "contributors_enabled": false,
            "created_at": "Thu Jan 03 22:19:25 +0000 2008",
            "default_profile": false,
            "default_profile_image": false,
            "description": "Diario de Mar del Plata. Edición digital https://t.co/u03KfGdfNC contacto@lacapitalmdq.com.ar",
            "entities": Object {
              "description": Object {
                "urls": Array [
                  Object {
                    "display_url": "LACAPITALMDP.COM",
                    "expanded_url": "http://WWW.LACAPITALMDP.COM",
                    "indices": Array [
                      41,
                      64,
                    ],
                    "url": "https://t.co/u03KfGdfNC",
                  },
                ],
              },
              "url": Object {
                "urls": Array [
                  Object {
                    "display_url": "lacapitalmdp.com",
                    "expanded_url": "http://www.lacapitalmdp.com",
                    "indices": Array [
                      0,
                      22,
                    ],
                    "url": "http://t.co/4292tDBdsg",
                  },
                ],
              },
            },
            "favourites_count": 28,
            "follow_request_sent": false,
            "followers_count": 77746,
            "following": true,
            "friends_count": 381,
            "geo_enabled": true,
            "has_extended_profile": false,
            "id": 11816412,
            "id_str": "11816412",
            "is_translation_enabled": false,
            "is_translator": false,
            "lang": "es",
            "listed_count": 384,
            "location": "Mar del Plata, Argentina",
            "name": "La Capital de MDP",
            "notifications": false,
            "profile_background_color": "DBE9ED",
            "profile_background_image_url": "http://abs.twimg.com/images/themes/theme17/bg.gif",
            "profile_background_image_url_https": "https://abs.twimg.com/images/themes/theme17/bg.gif",
            "profile_background_tile": true,
            "profile_banner_url": "https://pbs.twimg.com/profile_banners/11816412/1455119119",
            "profile_image_url": "http://pbs.twimg.com/profile_images/697354376910917632/gfgbLNPG_normal.png",
            "profile_image_url_https": "https://pbs.twimg.com/profile_images/697354376910917632/gfgbLNPG_normal.png",
            "profile_link_color": "CC3366",
            "profile_sidebar_border_color": "000000",
            "profile_sidebar_fill_color": "E6F6F9",
            "profile_text_color": "333333",
            "profile_use_background_image": true,
            "protected": false,
            "screen_name": "lacapitalmdq",
            "statuses_count": 78148,
            "time_zone": null,
            "translator_type": "none",
            "url": "http://t.co/4292tDBdsg",
            "utc_offset": null,
            "verified": false,
          },
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReached={[Function]}
    onEndReachedThreshold={5}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</Styled(ViewNB)>
`;
